src/Raven.Server/Utils/Metrics/MeterMetric.cs

```diff
@@ -131,42 +104,167 @@ public void Mark()
+
+                buckets.Add((start, count / bucketSeconds));
+            }
+
+            buckets.Sort((a, b) => a.Start.CompareTo(b.Start));
+
+            // Emit raw 100ms samples so diagnostics can plot the exact workload shape (no smoothing).
+            var raw = new DynamicJsonValue();
+            foreach (var (start, rate) in buckets)
+            {
+                var ageSeconds = (nowNs - start) / (double)Clock.NanosecondsInSecond;
+                var timestamp = nowUtc - TimeSpan.FromSeconds(ageSeconds);
+                raw[timestamp.ToString("O", CultureInfo.InvariantCulture)] = Math.Round(rate, 1);
+            }
+
+            result["Raw"] = raw;
+            return result;
+        }
+
+        // A small CAS loop guards bucket advancement to avoid the "clear-after-publish" race.
+        // Protocol:
+        // - Writers claim a bucket by CAS-ing Start to a sentinel value (ClaimedStart).
+        // - Under the claim, we zero Count/DurationSum, then publish the real Start via a release write.
+        // - Readers acquire Start and skip buckets with Start <= 0 (never used or claimed/in-progress).
+        // This ensures no thread observes half-initialized state. The loop runs only at quantum boundaries.
+        private void AddToBucket(long timestamp, long value, long duration)
+        {
+            var quantum = timestamp / BucketDurationNanoseconds;
+            var bucketStart = quantum * BucketDurationNanoseconds;
+            var index = (int)(quantum % BucketCount);
+
+            ref var bucket = ref _buckets[index];
+
+            const long ClaimedStart = -1; // sentinel: bucket claimed for reinitialization
+
+            while (true)
             {
-                var d = _m15Rate[i];
-                if (filterEmpty == false || Math.Abs(d) > double.Epsilon)
-                    results[now.ToString()] = Math.Round(d, 1);
-                now = now - oneSec;
+                var start = Volatile.Read(ref bucket.Start);
+                if (start == bucketStart)
+                    break; // already advanced for this quantum
+
+                if (start > 0 && bucketStart < start)
+                {
+                    // Another thread advanced further; follow it to avoid going backward.
+                    bucketStart = start;
+                    break;
+                }
+
+                if (start == ClaimedStart)
+                    continue; // someone else is initializing; wait until they publish
+
+                // Try to claim for reinitialization
+                if (Interlocked.CompareExchange(ref bucket.Start, ClaimedStart, start) != start)
+                    continue; // Lost the race; retry
+
+                // Own the bucket under the claim: zero fields, then publish with release semantics
+                Volatile.Write(ref bucket.Count, 0);
+                Volatile.Write(ref bucket.DurationSum, 0);
+                Volatile.Write(ref bucket.Start, bucketStart);
```

Comment:
You need only last volatile. You own the slot.

```suggestion
                bucket.Count = 0;
                bucket.DurationSum = 0;
                Volatile.Write(ref bucket.Start, bucketStart);
```